auslib/web/public/client.py

```diff
@@ -93,7 +93,31 @@ def getQueryFromURL(url):
     return query
 
 
+def extract_query_version(request_url):
+    version = 0
+    pattern = '^.*/update/([1-6])/.*\.xml$'
+    match = re.match(pattern, request_url)
+    if match:
+        version = int(match.group(1))
+    return version
+
+
+def update_query_version(request_url, url_params):
```

Comment:
Could you rework this to be a pure function (return the query version instead of updating an input parameter)? This makes the code more readable and testable.